### PR TITLE
Add publish script.

### PR DESCRIPTION
**Pull request by amandeepg**
Using the [bintray-release](https://github.com/novoda/bintray-release), along with their instructions for [publishing to Maven Central](https://github.com/novoda/bintray-release/wiki/Add-support-for-syncing-to-maven-central)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -21,8 +21,8 @@ buildscript {
         jcenter()
     }
     dependencies {
-        // NOTE: Do not place your application dependencies here; they belong
-        // in the individual module build.gradle files
+        classpath 'com.android.tools.build:gradle:2.3.1'
```
Comment (efrohnhoefer) [inline build.gradle]: Is this still required for a Java only project?

File: serialization/publish.gradle [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    version = project.VERSION_NAME
+
+    gradle.taskGraph.whenReady { taskGraph ->
+        taskGraph.getAllTasks().find{it.path == ":$project.name:generatePomFileForMavenPublication"}.doLast {
+            file("build/publications/maven/pom-default.xml").delete()
+            pom {
+                project {
+                    name "Serial"
+                    artifactId "serial"
+                    packaging project.name == 'compiler' ? 'jar' : 'aar'
```
Comment (efrohnhoefer) [inline serialization/publish.gradle]: I think we can just use `packaging jar`.

File: serialization/publish.gradle [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                            url 'https://www.twitter.com/'
+                        }
+                    }
+                }
+            }.writeTo("build/publications/maven/pom-default.xml")
+        }
+    }
+}
+
+publish {
```
Comment (efrohnhoefer) [inline serialization/publish.gradle]: All these values can be moved to gradle.properties

File: serialization/publish.gradle [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    repoName = 'Serial'
+    groupId = 'com.twitter.serial'
+    artifactId = 'serial'
+    publishVersion = project.VERSION_NAME
```
Comment (efrohnhoefer) [inline serialization/publish.gradle]: project.VERSION_NAME -> VERSION_NAME

File: serialization/publish.gradle (hunk 2 above)
Comment (amandeepg) [reply serialization/publish.gradle]: Based on our POM file here, https://repo1.maven.org/maven2/com/twitter/serial/serial/0.1.5/serial-0.1.5.pom, I think it's actually "packaging 'aar'" that we want. Agree that we don't need this ternary though

File: serialization/publish.gradle (hunk 3 above)
Comment (amandeepg) [reply serialization/publish.gradle]: Ah okay, I wasn't sure what should go there, so I just put stuff I expected would change, i.e. just the version name

File: build.gradle (hunk 1 above)
Comment (amandeepg) [reply build.gradle]: Good question, file an issue or send a PR for this to discuss separately, I'm not sure.

File: serialization/publish.gradle (hunk 2 above)
Comment (efrohnhoefer) [reply serialization/publish.gradle]: I think JAR is the correct value since the artifact is a jar file.

**Comment by efrohnhoefer** [comment]
LGTM 🚢

**Pull request merged**